src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.h

```diff
@@ -28,20 +30,91 @@ namespace crimson::os::seastore::omap_manager {
 
 struct OMapInnerNode
   : OMapNode,
-    StringKVInnerNodeLayout {
+    StringKVInnerNodeLayout,
+    ParentNode<OMapInnerNode, std::string>,
+    ChildNode<OMapInnerNode, OMapInnerNode, std::string> {
   using OMapInnerNodeRef = TCachedExtentRef<OMapInnerNode>;
-  using internal_iterator_t = const_iterator;
+  using internal_const_iterator_t = const_iterator;
+  using internal_iterator_t = iterator;
+  using parent_node_t = ParentNode<OMapInnerNode, std::string>;
+  using base_child_t = BaseChildNode<OMapInnerNode, std::string>;
+  using child_node_t = ChildNode<OMapInnerNode, OMapInnerNode, std::string>;
+  static constexpr uint32_t CHILD_VEC_UNIT = 128;
 
   explicit OMapInnerNode(ceph::bufferptr &&ptr)
-    : OMapNode(std::move(ptr)) {
-    this->set_layout_buf(this->get_bptr().c_str());
-  }
+    : OMapNode(std::move(ptr)),
+      StringKVInnerNodeLayout(get_bptr().c_str()),
+      parent_node_t(p2roundup(get_size(), CHILD_VEC_UNIT)) {}
   // Must be identical with OMapInnerNode(ptr) after on_fully_loaded()
   explicit OMapInnerNode(extent_len_t length)
-    : OMapNode(length) {}
+    : OMapNode(length),
+      StringKVInnerNodeLayout(nullptr),
+      parent_node_t(CHILD_VEC_UNIT) {}
```

Comment:
For the dynamic capacity in omap tree, I think it might be good to start with `parent_node_t(0)`, because `on_fully_loaded()` is supposed to be called upon read done, which means `capacity` will be guaranteed to be set to a reasonable value before use.